openmmforcefields/generators/template_generators.py

```diff
@@ -1705,14 +1726,16 @@ def generate_residue_template(self, molecule, residue_atoms=None):
         regenerate_impropers(molecule_graph)
 
         # Assign parameters
+        # NOTE: espaloma (nn) partial charges are assigned to molecules automatically if available.
+        # We need to overwrite the partial charges if we want to read them from the molecule.
```

Comment:
Good catch! We need to make sure if this is the case because we want users to be able to use charges from other methods/models combined with espaloma. I'll raise an issue about this so we can keep track of this.